### PR TITLE
Re-enable phone login

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: app/script/view_model/AuthViewModel.coffee [hunk 1]
```diff
@@ -745,21 +795,39 @@ class z.ViewModel.AuthViewModel
...
       {context: z.auth.AuthView.MODE.ACCOUNT_PHONE}
 
+  show_account_register: (focus = 'wire-register-name') ->
+    switch_params =
+      section: z.auth.AuthView.SECTION.ACCOUNT
+      mode: z.auth.AuthView.MODE.ACCOUNT_REGISTER
+      focus: focus
+    @switch_ui switch_params
+    amplify.publish z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.REGISTRATION.OPENED_EMAIL_SIGN_UP,
+      {context: @registration_context}
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee]:
```
context: @registration_context
```

File: app/script/view_model/AuthViewModel.coffee [hunk 2]
```diff
@@ -745,21 +795,39 @@ class z.ViewModel.AuthViewModel
     amplify.publish z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.ACCOUNT.OPENED_LOGIN,
       {context: z.auth.AuthView.MODE.ACCOUNT_PHONE}
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee]: here as well.. no need for `{`

File: app/script/view_model/AuthViewModel.coffee [hunk 3]
```diff
@@ -707,33 +766,24 @@ class z.ViewModel.AuthViewModel
...
     switch_params =
       section: z.auth.AuthView.SECTION.ACCOUNT
       mode: z.auth.AuthView.MODE.ACCOUNT_LOGIN
-      method: z.auth.AuthView.MODE.ACCOUNT_EMAIL
-      focus: 'wire-login-email'
+      focus: 'wire-login-username'
     @switch_ui switch_params
     amplify.publish z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.ACCOUNT.OPENED_LOGIN,
-      {context: z.auth.AuthView.MODE.ACCOUNT_EMAIL}
+      {context: @visible_method()}
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee]:
```
context: @visible_method()
```

File: app/script/view_model/AuthViewModel.coffee [hunk 4]
```diff
@@ -413,45 +397,106 @@ class z.ViewModel.AuthViewModel
...
+          remember_me: @persist()
+        }
+      @_authentication_successful()
+    .catch (error) =>
+      @pending_server_request false
+      $('#wire-verify-password').focus()
+      if navigator.onLine
+        if error.label is z.service.BackendClientError::LABEL.PENDING_ACTIVATION
+          return @_set_hash z.auth.AuthView.MODE.POSTED_PENDING
+        else if error.label
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee]: `else` is redundant..

File: app/script/view_model/AuthViewModel.coffee [hunk 5]
```diff
@@ -413,45 +397,106 @@ class z.ViewModel.AuthViewModel
...
+
+    @pending_server_request true
+    payload = @_create_payload z.auth.AuthView.MODE.VERIFY_PASSWORD
+    @auth.repository.login payload, @persist()
+    .then =>
+      amplify.publish z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.ACCOUNT.LOGGED_IN,
+        {
+          context: z.auth.AuthView.MODE.VERIFY_PASSWORD
+          remember_me: @persist()
+        }
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee]: no need for `{}`

File: app/script/view_model/AuthViewModel.coffee [hunk 6]
```diff
@@ -413,45 +397,106 @@ class z.ViewModel.AuthViewModel
...
+
+    @pending_server_request true
+    payload = @_create_payload z.auth.AuthView.MODE.VERIFY_CODE
+    @auth.repository.login payload, @persist()
+    .then =>
+      amplify.publish z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.ACCOUNT.LOGGED_IN,
+        {
+          context: z.auth.AuthView.MODE.ACCOUNT_PHONE
+          remember_me: @persist()
+        }
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee]: `{}`

File: app/script/view_model/AuthViewModel.coffee [hunk 7]
```diff
@@ -272,39 +274,17 @@ class z.ViewModel.AuthViewModel
...
-    payload = @_create_payload z.auth.AuthView.MODE.ACCOUNT_EMAIL
+    payload = @_create_payload z.auth.AuthView.MODE.ACCOUNT_PASSWORD
     @auth.repository.login payload, @persist()
     .then =>
       amplify.publish z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.ACCOUNT.LOGGED_IN,
         {
-          context: z.auth.AuthView.MODE.ACCOUNT_EMAIL
+          context: z.auth.AuthView.MODE.ACCOUNT_PASSWORD
           remember_me: @persist()
         }
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee]: `{}`

File: app/script/view_model/AuthViewModel.coffee [hunk 8]
```diff
@@ -106,9 +107,9 @@ class z.ViewModel.AuthViewModel
...
     @code_expiration_timestamp.subscribe (timestamp) =>
       @code_expiration_in moment.unix(timestamp).fromNow()
-      @code_interval_id = setInterval =>
+      @code_interval_id = window.setInterval =>
         if timestamp <= z.util.get_unix_timestamp()
-          clearInterval @code_interval_id
+          window.clearInterval @code_interval_id
           return @code_expiration_timestamp 0
         @code_expiration_in moment.unix(timestamp).fromNow()
       , 20000
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee:115]: Magic number.. maybe constant?

File: app/script/view_model/AuthViewModel.coffee [hunk 9]
```diff
@@ -67,11 +67,12 @@ class z.ViewModel.AuthViewModel
     @client_type = ko.observable z.client.ClientType.TEMPORARY
     @country_code = ko.observable ''
     @country = ko.observable ''
-    @phone_number = ko.observable ''
-    @email = ko.observable ''
     @name = ko.observable ''
     @password = ko.observable ''
     @persist = ko.observable if z.util.Environment.electron then true else false
```
Comment (lipis) [inline app/script/view_model/AuthViewModel.coffee]: not part of the PR.. but since we are here.. could be just..

``` coffeescript
@persist = ko.observable z.util.Environment.electron
```

File: app/style/auth/verify.less [hunk 10]
```diff
@@ -40,9 +40,10 @@
       margin: 0 0 32px 0;
     }
   }
-  #form-verify-email {
-    .verify-email-text {
-      margin: 16px 8px 36px 0px;
+
+  #form-verify-phone-password {
+    .verify-password-text {
+      margin: 0 0 32px 0;
```
Comment (lipis) [inline app/style/auth/verify.less]:
``` css
margin-bottom: 32px;
```

File: app/style/auth/account.less [hunk 11]
```diff
@@ -45,13 +45,12 @@
...
-    min-height: 32px;
+    min-height: 52px;
   }
 
   #form-account-login {
     .selector-method {
-      opacity: 0; /* TODO: remove when re-enabling phone login */
-      height: 0px; /* TODO: reset to 24px when re-enabling phone login */
+      height: 24px;
       margin: 0 72px 24px 72px;
```
Comment (lipis) [inline app/style/auth/account.less]:
``` css
margin: 0 72px 24px;
```

File: app/style/auth/verify.less (hunk 10 above)
Comment (gregor) [reply app/style/auth/verify.less]: No. I am overwriting some other values.

Comment (lipis) [reply app/style/auth/verify.less]: or at least that if it's indeed overriding the top/left/right :)

```
margin: 0 0 32px;
```
